incubator/hnc/internal/reconcilers/object.go

```diff
@@ -659,8 +658,8 @@ func (r *ObjectReconciler) syncPropagation(ctx context.Context, log logr.Logger,
 // - Objects have a selector that doesn't match the destination namespace
 // - Service Account token secrets
 func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstructured.Unstructured, dst string) bool {
-	selector := r.getSelector(log, inst)
-	treeSelector := r.getTreeSelector(log, inst)
+	selector, _ := selectors.GetSelector(log, inst)
+	treeSelector, _ := selectors.GetTreeSelector(log, inst)
```

Comment:
Throwing away return values is generally a "code smell" - not _necessarily_ an error, but a good indication that you should probably have a much closer look at what you're throwing away. Very likely, the thing you're throwing away isn't valuable (in which case, why are you returning it in the first place?) or it _is_ valuable (in which case, why are you throwing it away?) - in this case, I think the former applies.

Reply:
My understanding is that because in 'validators' we need the error to show it to the user, but here we have already logged the error so we don't need it in the 'reconcilers' anymore.

Reply:
Right, but see my other comments about where the error _should_ be logged.

Reply:
Done